lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -0,0 +1,141 @@
+from opaque_keys.edx import locator
+import unittest
+import httpretty
+from urlparse import urljoin
+
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from student.models import CourseEnrollment
+from openedx.core.djangoapps.programs.tests.mixins import ProgramsApiConfigMixin
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from lms.djangoapps.learner_dashboard.tests.mixins import ProgramsDataMixin
+from oauth2_provider.tests.factories import ClientFactory
+from provider.constants import CONFIDENTIAL
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestProgramListing(
+        ModuleStoreTestCase,
+        ProgramsApiConfigMixin,
+        ProgramsDataMixin):
+
+    """
+    Unit tests for getting the list of programs enrolled by a logged in user
+    """
+    PASSWORD = 'test'
+
+    def setUp(self):
+        """
+        Add a student
+        """
+        super(TestProgramListing, self).setUp()
+        ClientFactory(name=ProgramsApiConfig.OAUTH2_CLIENT_NAME, client_type=CONFIDENTIAL)
+        self.student = UserFactory()
+        self.student.set_password(self.PASSWORD)
+        self.student.save()
+        self.create_programs_config()
+
+    def _create_course_and_enroll(self, student, org, course, run):
+        """ Creates a course and associated enrollment. """
+        course_location = locator.CourseLocator(org, course, run)
+        course = CourseFactory.create(
+            org=course_location.org,
+            number=course_location.course,
+            run=course_location.run
+        )
+        enrollment = CourseEnrollment.enroll(student, course.id)
+        enrollment.created = datetime.datetime(2000, 12, 31, 0, 0, 0, 0)
+        enrollment.save()
+
+    @override_settings(MKTG_URLS={'ROOT': 'http://edx.org'})
+    def _get_program_url(self, marketing_slug):
+        '''
+            Helper function to get the program card url
+        '''
+        return urljoin(
+            settings.MKTG_URLS.get('ROOT'),
+            'xseries' + '/{}'
+        ).format(marketing_slug)
+
+    @override_settings(MKTG_URLS={'ROOT': 'http://edx.org'})
+    def _setup_and_get_program(self):
+        '''
+            The core function to setup the mock program api,
+            then call the django test client to get the actual program listing page
+            make sure the request suceeds and make sure x_series_url is on the page
+        '''
+        self.mock_programs_api()
+        self.client.login(username=self.student.username, password=self.PASSWORD)
+        response = self.client.get(reverse("program_listing_view"))
+        self.assertEqual(response.status_code, 200)
+        x_series_url = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries')
+        self.assertIn(x_series_url, response.content)
```

Comment:
Upon further investigation, I am leaving this as it is. The check is to make sure the URL is passed from the server to the web page in a json format. Afterwords, the backbone app will make sure it is properly displayed in the DOM